spot-ml/src/main/scala/org/apache/spot/proxy/ProxySuspiciousConnectsModel.scala

```diff
@@ -118,35 +112,20 @@ object ProxySuspiciousConnectsModel {
     logger.info("training new proxy suspcious connects model")
 
 
-    val selectedRecords = inputRecords.select(Date, Time, ClientIP, Host, ReqMethod, UserAgent, ResponseContentType, RespCode, FullURI)
+    val selectedRecords =
+      inputRecords.select(Date, Time, ClientIP, Host, ReqMethod, UserAgent, ResponseContentType, RespCode, FullURI)
       .unionAll(ProxyFeedback.loadFeedbackDF(sparkContext, sqlContext, config.feedbackFile, config.duplicationFactor))
 
-    val timeCuts =
-      Quantiles.computeDeciles(selectedRecords
-        .select(Time)
-        .rdd
-        .flatMap({ case Row(t: String) =>
-          Try {
-            TimeUtilities.getTimeAsDouble(t)
-          } match {
-            case Failure(_) => Seq()
-            case Success(time) => Seq(time)
-
-          }
-        }))
-
-    val entropyCuts = Quantiles.computeQuintiles(selectedRecords
-      .select(FullURI)
-      .rdd
-      .flatMap({ case Row(uri: String) =>
-        Try {
-          Entropy.stringEntropy(uri)
-        } match {
-          case Failure(_) => Seq()
-          case Success(entropy) => Seq(entropy)
-        }
-
-      }))
+    // These buckets are optimized to datasets used for training. Last bucket is of larger size to ensure fit.
+    // The maximum value of entropy is given by log k where k is the number of distinct categories.
+    // Given that the alphabet and number of characters is finite the maximum value for entropy is upper bounded.
+    // Assuming 10,000 distinct categories the entropy is bounded by 13.38. Empirical tests showed a maximum
+    // value of 6.2 in our training set.
+    // Bucket number and size can be changed to provide less/more granularity
+    val entropyCuts = Array(0.0, 0.3, 0.6, 0.9, 1.2,
+      1.5, 1.8, 2.1, 2.4, 2.7,
+      3.0, 3.3, 3.6, 3.9, 4.2,
+      4.5, 4.8, 5.1, 5.4, 20)
```

Comment:
personally I prefer having a last bucket that trails off to positive infinity, even if  it would take a uniform distribution on millions of characters to achieve entropy of 20... I just like knowing that the corner cases are handled  (although it is a theoretical concern since there are only so many characters possible in a url)